src/shared/Core.Tests/InputArgumentsTests.cs

```diff
@@ -176,6 +176,28 @@ public void InputArguments_GetRemoteUri_AuthorityAndPort_ReturnsUriWithAuthority
             Assert.Equal(expectedUri, actualUri);
         }
 
+        [Theory]
+        [InlineData("example.com", 456, "foo")]
+        [InlineData("example.com", 789, "foo/bar")]
+        [InlineData("example.foo.com", 789, "bar")]
+        public void InputArguments_GetRemoteUri_AuthorityAndPortAndPath_ReturnsUriWithAuthorityAndPortAndPath(string authority, int port, string path)
```

Comment:
@mjcheetham - just a heads up that I used `Authority` here to align with the names of the other tests in this file.